frontend/public/components/secrets/create-secret.tsx

```diff
@@ -87,7 +88,7 @@ const generateSecret = () => {
 };
 
 // withSecretForm returns SubForm which is a Higher Order Component for all the types of secret forms.
-const withSecretForm = (SubForm) => class SecretFormComponent extends React.Component<BaseEditSecretProps_, BaseEditSecretState_> {
+export const withSecretForm = (SubForm, renderinModal?:boolean) => class SecretFormComponent extends React.Component<BaseEditSecretProps_, BaseEditSecretState_> {
```

Comment:
Typo `renderinModal`. I suggest you simply use the name `modal`

```suggestion
export const withSecretForm = (SubForm, modal?: boolean) => class SecretFormComponent extends React.Component<BaseEditSecretProps_, BaseEditSecretState_> {
```

Reply:
Done